### ISSUE TITLE
SkLearn Logistic Regression behaves weirdly if the learning rate is not set to constant.

### ISSUE THREAD

**Issue by alitolga** [fedbiomed/fedbiomed#1632]
In Scikit-Learn FedSGDClassifier the training loss oscillates a lot if the `learning_rate` is not set to `constant` in the `model_args`.

The issue is still to be investigated, but the likely cause is the learning rate getting reset at the beginning of each round, which confuses the model during `partial_fit`.

**Comment by ybouilla** [comment]
Hi Ali, I think you guessed right, actually scikit-learn doesnot provide a way to retrieve learning rate after n iterations, so the one saved between 2 rounds is the one at epoch 0 (first iteration). You  will have the same issue for FedSGDClassifier.
From what i remember, it' s very hard to retrieve a non-constant learning rate of a scikit-learn model. The only way to fix that is to compute sperately the learning rate the way scikit-learn compute it, and then store it and retieve it between 2 rounds. This hasnot been implemented because at that time, we wanted the user to prefer using `declearn` optimizers instead, and `declearn` learning rate schedulers,

**Comment by alitolga** [comment]
Thanks Yannick! That confirms the suspicion.

So then, I guess it doesn't make sense to scikit-learn optimizers without a constant learning rate.

**Comment by ybouilla** [comment]
I think you still can but you will have to adapt given an heuristic ... and to adjust it at the end of each round. it looks it s way too complicated to implement for what it will bring as benefits. For your use case, don't you want to use a declearn optimizer? They are more suitable for Federated learning training
